app/api/v2/__init__.py

```diff
@@ -1,15 +1,51 @@
 from __future__ import annotations
 
+from typing import Any
+
 from fastapi import APIRouter
+from fastapi import Depends
+from fastapi import HTTPException
+from fastapi import status
+
+from app.api.v2.common.oauth import OAuth2Scheme
+from app.repositories import access_tokens as access_tokens_repo
+
+
+oauth2_scheme = OAuth2Scheme(
+    authorizationUrl="/v2/oauth/authorize",
+    tokenUrl="/v2/oauth/token",
+    refreshUrl="/v2/oauth/refresh",
+    scheme_name="OAuth2 for third-party clients.",
+    scopes={
+        "public": "Access endpoints with public data.",
+        "identify": "Access endpoints with user's data.",
+        "admin": "Access admin endpoints.",
+    },
+)
+
+
+async def get_current_client(token: str = Depends(oauth2_scheme)) -> dict[str, Any]:
```

Comment:
I tried to do this with the inner function but can't pass the arguments to the function in `Depends`.

Reply:
np